packages/dd-trace/test/appsec/iast/analyzers/nosql-injection-mongodb-analyzer.mongoose.plugin.spec.js

```diff
@@ -52,91 +52,187 @@ describe('nosql injection detection in mongodb - whole feature', () => {
-            fn: async (req, res) => {
-              Test.find({
-                name: {
-                  child: [req.query.key]
-                }
-              }).then(() => {
-                res.end()
-              })
-            },
-            vulnerability: 'NOSQL_MONGODB_INJECTION',
-            makeRequest: (done, config) => {
-              axios.get(`http://localhost:${config.port}/?key=value`).catch(done)
-            }
-          })
+          describe('using promises', () => {
+            testThatRequestHasVulnerability({
+              fn: async (req, res) => {
+                Test.find({
+                  name: req.query.key,
+                  value: [1, 2,
+                    'value',
+                    false, req.query.key]
+                }).then(() => {
+                  res.end()
+                })
+              },
+              vulnerability: 'NOSQL_MONGODB_INJECTION',
+              makeRequest: (done, config) => {
+                axios.get(`http://localhost:${config.port}/?key=value`).catch(done)
+              }
+            })
 
-          testThatRequestHasVulnerability({
-            testDescription: 'should have NOSQL_MONGODB_INJECTION vulnerability in correct file and line',
-            fn: async (req, res) => {
-              const filter = {
-                name: {
-                  child: [req.query.key]
-                }
+            testThatRequestHasVulnerability({
+              fn: async (req, res) => {
+                Test.find({
+                  name: {
+                    child: [req.query.key]
+                  }
+                }).then(() => {
+                  res.end()
+                })
+              },
+              vulnerability: 'NOSQL_MONGODB_INJECTION',
+              makeRequest: (done, config) => {
+                axios.get(`http://localhost:${config.port}/?key=value`).catch(done)
               }
-              require(tmpFilePath)(Test, filter, () => {
-                res.end()
-              })
-            },
-            vulnerability: 'NOSQL_MONGODB_INJECTION',
-            makeRequest: (done, config) => {
-              axios.get(`http://localhost:${config.port}/?key=value`).catch(done)
-            },
-            occurrences: {
-              occurrences: 1,
-              location: {
-                path: vulnerableMethodFilename,
-                line: 4
+            })
+
+            testThatRequestHasVulnerability({
+              fn: async (req, res) => {
+                Test.find({
+                  name: {
+                    child: [req.query.key]
+                  }
+                }).then(() => {
+                  res.end()
+                })
+              },
+              vulnerability: 'NOSQL_MONGODB_INJECTION',
+              makeRequest: (done, config) => {
+                axios.get(`http://localhost:${config.port}/?key=value`).catch(done)
               }
-            }
-          })
+            })
```

Comment:
this is an exact duplicate of the test above no ?

Reply:
deleted